From 68a72b4938cdaf691b707f20a20f6a457516f200 Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Tue, 31 May 2016 08:09:05 +0530
Subject: [PATCH] EM/power_supply_charger: Fixing the KW issue

Issue:
KW throws error for the statement SET_MAX_THROTTLE_STATE is called
with psy object as NULL.

Solution:
Splitting the combined the if condition check in
power_supply_register_charger function and call SET_MAX_THROTTLE_STATE
only if psy object is not NULL.

Change-Id: I7369cd6dd4bc3153757a90c6ea4f53ab648eeb11
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-32961
Reviewed-on: https://android.intel.com:443/511078
---
 drivers/power/power_supply_charger.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index a4b548e..38d1c9e 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -1017,12 +1017,14 @@ int power_supply_register_charger(struct power_supply *psy)
 {
 	int ret = 0;
 
-	if (!psy_chrgr.is_cable_evt_reg && psy) {
-		ret = register_usb_notifier();
-		schedule_work(&notifier_work);
+	if (psy) {
+		if (!psy_chrgr.is_cable_evt_reg) {
+			ret = register_usb_notifier();
+			schedule_work(&notifier_work);
+		}
+		SET_MAX_THROTTLE_STATE(psy);
 	}
 	psy_chrgr.is_cable_evt_reg = true;
-	SET_MAX_THROTTLE_STATE(psy);
 	return ret;
 }
 EXPORT_SYMBOL(power_supply_register_charger);
-- 
1.9.1

